File: AST/StatementList.java
package AST;

import java.util.List;

import AST.Visitor.Visitor;

import java.util.ArrayList;
import java_cup.runtime.ComplexSymbolFactory.Location;

public class StatementList extends Statement {
   private List<Statement> list;

   public StatementList(Location pos) {
      super(pos);
      list = new ArrayList<Statement>();
   }

   public void add(Statement n) {
      list.add(n);
   }

   public Statement get(int i)  { 
      return list.get(i); 
   }

   public int size() { 
      return list.size(); 
   }

  public void accept(Visitor v) {
    v.visit(this);
  }
}

File: AST/StatementList.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static Antlr4.Runtime.Atn.SemanticContext;

namespace AST
{
    // not in final AST tree, but useful immediate node for generating it
    public class StatementList : AstNode, Statement, Printable, ProgramPiece
    {
        public List<AstNode> Statements;

        public StatementList(List<AstNode> statements, int lineNumber, int startCol)
        {
            Statements = statements;
            LineNumber = lineNumber;
            StartCol = startCol;
        }

        public override void Accept(Visitor v) { }
    }
}
